### DISCUSSION TITLE
How can I conditionally install a library based on a condition in manifest mode?

### DISCUSSION THREAD

**Discussion by eladmaimoni** [microsoft/vcpkg#28171] [Q&A]
I have a CMake project that used vcpkg.json manifest file. 
I would like some of the libraries to install only if some conditions are met. For example only if a certain cmake variable is set to "ON". 

There is the ["platform" field](https://vcpkg.readthedocs.io/en/latest/users/manifests/#platform) in the manifest file that seems like it has this functionality. But I could not find a documentation as to how I can define my own "platform" for use cases other than "install this only if we are not on windows".

How can I conditionally install a library based on a condition in manifest mode?

**Comment by autoantwort** [comment]
See https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/cmake-integration.md#vcpkg_manifest_features

**Reply by eladmaimoni** [reply]
@autoantwort nice. thank you. but isn't the CmakeLists.txt runs only after vcpkg already installs the dependencies? how does that work exactly?

**Reply by autoantwort** [reply]
> but isn't the CmakeLists.txt runs only after vcpkg already installs the dependencies

No
